app/locale/en.ozar.coffee

```diff
@@ -3119,6 +3119,50 @@ module.exports = nativeDescription: "English", englishDescription: "English", tr
     are_you_missing_a: "Are you missing a '$1' after '$2'? "
     your_parentheses_must_match: "Your parentheses must match."
 
+    error_index_empty: "Cannot read property `__name__` of `__specTypeName__`"
+    error_not_iterable: "`__var__` is not iterable."
+    error_assignment_invalid: "Invalid reference in assignment."
+    error_invalid_array_length: "Invalid array length."
+    error_didnt_make_generator: "`__name__` didnt make a generator."
+    error_undefined_variable: "`__name__` is not defined."
+    error_declared_identifier: "Identifier `__name__` has already been declared."
+    error_declarations_cant_resolve: "Couldnt resolve declarations component: `__type__`."
+    error_strict_no_statement: "Strict mode code may not include a with statement."
+    error_cant_resolve_ref_component: "Couldnt resolve ref component: `__type__`."
+    error_cant_convert_null: "Cannot convert undefined or null to object."
+    error_need_object: "Need an object."
+    error_illegal_object_prototype: "Object prototype may only be an Object or null."
+    error_no_prototype: "No prototype."
+    error_cant_write_property_to_undefined: "Can't write property of undefiend: `__idx__`."
+    error_cant_write_property_to_non_obj: "Can't write property of non-object type: `__idx__`."
+    error_function_isnt_constructor: "function is not a constructor"
+    error_write_protected_property: "Can't write to protected property: `__name__`"
+    error_read_protected_property: "Can't read protected property: `__name__`"
+    error_convert_value_failed: "Cannot convert `__orig__` value to `__new__` value."
+    error_convert_object_to_primitive: "Cannot convert object to primitive value."
+    error_first_argument_to_be_function: "First argument to `__fun__` must be a function."
+    error_write_non_writable: "Can't write to non-writable value."
+    error_delete_nonconfigurable_object: "Can't delete nonconfigurable object."
+    error_cant_access_get: "Can't access get `__name__` of that type."
+    error_base_obj_not_invokable: "Base object not invokeable."
+    error_arg2_not_callable: "Arg2 not callalble."
+    error_reduce_empty_array_without_init: "Reduce an empty array with no initial value."
+    error_realm_prototype_not_realm: "Realm prototype not realm."
+    error_function_tostring_not_generic: "Function.prototype.toString is not generic"
+    error_regex_on_non_regex: "Calling regex method on non regex.",
+    error_call_string_on_null: "called String function on null or undefined?"
+    error_wrong_type: "Wrong Type."
+    error_replace_callback_not_written: "Replace with callbacks not written yet"
+    error_right_hand_side_not_callable: "Right-hand side of `__name__` is not callable"
+    error_segmentation_fault: "Segmentation Fault `__offset__`."
+    error_no_arg_to_reference: "No argument to refrence."
```

Comment:
There are some typos in Esper's errors that also get through to our errors: "callalble", "refrence", "didnt", "couldnt", "undefiend", "invokeable", "derefrence", and some inconsistencies with capitalization and punctuation. If we wanted to fix these, can we just rewrite these strings and leave the i18n keys alone, or should we fix them in Esper and make sure they are spelled/written correctly there before making these i18n keys?

Reply:
yeah that's my confusion too. actually we can just update the i18n values , and left the key. but we can update keys too. 
i just copy paste these typo from esper 😂  i don't know if there're any special meanings around this `refrence` 😂